[TensorExpr] Replace `ExprHandle` with `const Expr*` in `Substitute`.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35176 [TensorExpr] Cleanup implementation of alloc/free insertion.
* #35175 [TensorExpr] Factor out LoopNest::insertAllocFree.
* #35174 [TensorExpr] Methods name cleanup in LoopNest class.
* **#35173 [TensorExpr] Replace `ExprHandle` with `const Expr*` in `Substitute`.**



Differential Revision: [D20585577](https://our.internmc.facebook.com/intern/diff/D20585577)

Commit:
[TensorExpr] Replace `ExprHandle` with `const Expr*` in `Substitute.

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Replace `ExprHandle` with `const Expr*` in `Substitute`."




Differential Revision: [D20585577](https://our.internmc.facebook.com/intern/diff/D20585577)

[ghstack-poisoned]

nickgg reviewed (approved):
is it your plan to remove ExprHandle everywhere internal?

ZolotukhinM commented:
> is it your plan to remove ExprHandle everywhere internal?

Yes, even though I'm pursuing it less actively now.

Commit:
Update on "[TensorExpr] Replace `ExprHandle` with `const Expr*` in `Substitute`."




Differential Revision: [D20585577](https://our.internmc.facebook.com/intern/diff/D20585577)

[ghstack-poisoned]

Commit:
Rebase on "[TensorExpr] Replace `ExprHandle` with `const Expr*` in `Substitute`."




Differential Revision: [D20585577](https://our.internmc.facebook.com/intern/diff/D20585577)

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] Replace `ExprHandle` with `const Expr*` in `Substitute`."




Differential Revision: [D20585577](https://our.internmc.facebook.com/intern/diff/D20585577)

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] Replace `ExprHandle` with `const Expr*` in `Substitute`."




Differential Revision: [D20585577](https://our.internmc.facebook.com/intern/diff/D20585577)

[ghstack-poisoned]
